src/thunderbots/backend_output/main.cpp

```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+#include "geom/point.h"
+#include <ros/ros.h>
+#include <thunderbots_msgs/MovePrimitive.h>
+#include "ai/primitive/move_prim.h"
+#include "ai/primitive/primitive.h"
+#include "backend_output/grsim/grsim_backend.h"
+
+// Member variables we need to maintain state
+
+// A vector of primitives. It is cleared each tick, populated by the callbacks
+// that receive primitive commands, and is processed by the backend to send
+// the primitives to the system we have chosen (such as grSim, our radio, etc.)
+std::vector<Primitive> primitives;
+
+// Callbacks
+void movePrimUpdateCallback(const thunderbots_msgs::MovePrimitive::ConstPtr &msg)
+{
+    thunderbots_msgs::MovePrimitive move_prim_msg = *msg;
+    MovePrim move_prim = MovePrim(move_prim_msg);
+    primitives.push_back(move_prim);
+}
+
+int main(int argc, char** argv)
+{
+    // Init ROS node
+    ros::init(argc, argv, "backend_output");
+    ros::NodeHandle n;
+
+    // Create subscribers to topics we care about
+    ros::Subscriber move_prim_sub = n.subscribe("backend/move_prim", 1, movePrimUpdateCallback);
+
```

Comment:
Do we want to have these string "backend_output" and "backend/move_prim" hardcoded? I know this is very simple, but maybe we need to put these kinds of things in some sort of constants/names file. Some sort of database type thing in case the names of the strings change so that we can leave the variable names they refer to the same.

Reply:
Yes not a bad idea considering these "paths" need to be consistent across nodes / files. I think that's more appropriate to deal with when we get to finalizing the interfaces, so I'll change it then unless there are any objections.

Made and issue for it here: #6

Reply:
Yeah let's just keep an eye on it before it gets out of hand